connect/src/test/scala/org/apache/spark/sql/connect/planner/SparkConnectProtoSuite.scala

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql.connect.planner
+
+import org.scalatest.BeforeAndAfter
+
+import org.apache.spark.connect.proto
+import org.apache.spark.sql.catalyst.connect.expressions.DslString
+import org.apache.spark.sql.catalyst.connect.plans.DslLogicalPlan
+import org.apache.spark.sql.catalyst.expressions.AttributeReference
+import org.apache.spark.sql.catalyst.plans.PlanTest
+import org.apache.spark.sql.catalyst.plans.logical.{LocalRelation, LogicalPlan}
+import org.apache.spark.sql.test.SharedSparkSession
+import org.apache.spark.sql.types.{DataType, IntegerType}
+
+/**
+ * This suite is based on connect DSL and test that given same dataframe operations, whether
+ * connect could construct a proto plan that can be translated back, and after analyzed, be the
+ * same as Spark dataframe's generated plan.
+ */
+class SparkConnectProtoSuite extends SharedSparkSession
+  with SparkConnectPlanTest with PlanTest with BeforeAndAfter {
+
+  lazy val connectTestRelation = createLocalRelationProto(Map("id" -> IntegerType))
+
+  lazy val sparkTestRelation = LocalRelation(AttributeReference("id", IntegerType)())
+
+  override protected def beforeAll(): Unit = {
+    super.beforeAll()
+    setupTestData()
+  }
+
+  test("Basic select") {
+    val connectPlan = analyze(connectTestRelation.select("id".protoAttr))
+    val sparkPlan = sparkTestRelation.select("id").queryExecution.analyzed
+    comparePlans(connectPlan, sparkPlan, false)
+  }
+
+  private def analyze(plan: proto.Relation): LogicalPlan = {
+    spark.sessionState.executePlan(transform(plan)).analyzed
+  }
+
+  private def createLocalRelationProto(colNameTypeMap: Map[String, DataType]): proto.Relation = {
+    val localRelationBuilder = proto.LocalRelation.newBuilder()
+    // TODO: set data types for each local relation attribute one proto supports data type.
+    for ((k, v) <- colNameTypeMap) {
+      localRelationBuilder.addAttributes(
+        proto.Expression.Attribute.newBuilder().setName(k).build()
+      )
+    }
+    proto.Relation.newBuilder().setLocalRelation(localRelationBuilder.build()).build()
+  }
+
+  protected override def setupTestData(): Unit = {
```

Comment:
do we need this? They are lazy val and will be initialized propertly when accessed.